mobile/src/main/java/mycroft/ai/MainActivity.java

```diff
@@ -249,7 +307,11 @@ public void sendMessage(String msg) {
             mWebSocketClient.send(json);
 
         } catch (WebsocketNotConnectedException e) {
-            Toast.makeText(getApplicationContext(), getResources().getString(R.string.websocket_closed), Toast.LENGTH_SHORT).show();
+            if (retryCount > maximumRetries) {
+                Toast.makeText(getApplicationContext(), getResources().getString(R.string.websocket_closed), Toast.LENGTH_SHORT).show();
+            } else { // retry
+                sendMessage(msg, retryCount++);
```

Comment:
The value of `retryCount++` will never be used because it increments _after_ line 313 finishes executing.

Anyway, the whole try-catch should be wrapped in a loop, which we break out of if `send(json)` succeeds or the retry count is reached. Something like

    do {
        try {
            // ....
            break;
        } catch (WebsocketNotConnectedException ignored) {
            retryCount++;
        }
    } while (retryCount < maximumRetries);
    if (retryCount >= maximumRetries) {
        // Toast/log/whatever here
    }